main.cpp

```diff
@@ -302,90 +229,37 @@ struct Numeric
                 return *this;
             }
         }
-        else if( x < std::numeric_limits<Type>::epsilon() )
+        else if( static_cast<NumericType>(x) < std::numeric_limits<NumericType>::epsilon() )
```

Comment:
You should be comparing with DivisorType's epsilon, not NumericType's epsilon
- [ ] revise